test/t8_forest/t8_gtest_face_neighbors.cxx

```diff
@@ -0,0 +1,325 @@
+          //        - If E face f has neighbor E' face f', then
+          //             E' face f' must have neighbor E face f.
+
+          // Now checking for inner and boundary elements.
+
+          // Compute whether this element is a boundary element or not.
+          // An element is a boundary element if it lies on the tree boundary
+          // and if the corresponding tree face is at the domain boundary.
+          const bool is_root_boundary = scheme->element_is_root_boundary (tree_class, element, iface);
+          const int tree_face = scheme->element_get_tree_face (tree_class, element, iface);
+          const bool is_boundary_element
+            = is_root_boundary && t8_cmesh_tree_face_is_boundary (cmesh, cmesh_tree, tree_face);
+
+          if (!is_boundary_element) {
+            if (!is_ghost) {  // Local element
+              if (forest_is_uniform) {
+                // In a uniform forest we must have exactly 1 neighbor.
+                EXPECT_EQ (num_neighbors, 1)
+                  << "Inner local element should have exactly 1 neighbor, has " << num_neighbors << ".";
+              }
+              else {
+                // In an adaptive forest we have 1 or more neighbors.
+                EXPECT_GE (num_neighbors, 1)
+                  << "Inner local element should have at least 1 neighbor, has " << num_neighbors << ".";
+              }
+            }
+            else {  // Ghost element
+              if (forest_is_uniform) {
+                // In a uniform forest a ghost element has none or one neighbor.
+                EXPECT_TRUE (num_neighbors == 0 || num_neighbors == 1)
+                  << "Inner ghost element should have exactly 1 or 0 neighbors, has " << num_neighbors << ".";
+              }
+              else {
+                // In an adaptive forest a ghost element has 0 or more neighbors.
+                EXPECT_GE (num_neighbors, 0)
+                  << "Inner ghost element should have 0 or more neighbors, has " << num_neighbors << ".";
+              }
+            }
+          }
+          else {
+            EXPECT_EQ (num_neighbors, 0) << "Boundary element should have exactly 0 neighbors, has " << num_neighbors
+                                         << ".";
+          }
+
+          // Check that the neighbor of the neighbor is the original element.
+          for (int ineigh = 0; ineigh < num_neighbors; ++ineigh) {
+            const t8_element_t *neighbor = neighbor_leaves[ineigh];
+            const int dual_face = dual_faces[ineigh];
+            const t8_locidx_t neigh_index = element_indices[ineigh];
+
+            t8_debugf ("Checking neighbor element %p in (global) tree %li.\n", (void *) neighbor, gneigh_tree);
+            t8_debugf ("dual face is %i, index is %i\n", dual_face, neigh_index);
+
+            ASSERT_TRUE (scheme->element_is_valid (neigh_class, neighbor))
+              << "Neighbor element " << ineigh << " is not valid";
```

Comment:
Luisa and Eva could not compile this with an ASSERT_TRUE @holke

Reply:
@sandro-elsweijer Do you have the exact compile error?

Reply:
@Davknapp already added a fix, currently on a different branch, will get merged soon.